src/mono/browser/runtime/invoke-cs.ts

```diff
@@ -289,13 +308,20 @@ function bind_fn(closure: BindingClosure) {
             }
 
             // call C# side
-            invoke_sync_method(method, args);
             if (is_async) {
+                invoke_async_jsexport(method, args, 2 + args_count);
                 // in case the C# side returned synchronously
                 js_result = end_marshal_task_to_js(args, undefined, js_result);
             }
-            else if (res_converter) {
-                js_result = res_converter(args);
+            else if (is_oneway) {
+                // call C# side, fire and forget
+                invoke_async_jsexport(method, args, 2 + args_count);
```

Comment:
if part of the oneway contract is 'there's no return value' it's probably worth asserting that there's no return value, otherwise you could end up with leaks

Reply:
It would not compile for non `void` signature. Or do you mean something else ?